go.mod

```diff
@@ -256,3 +256,5 @@ replace (
 	k8s.io/sample-cli-plugin => ./staging/src/k8s.io/sample-cli-plugin
 	k8s.io/sample-controller => ./staging/src/k8s.io/sample-controller
 )
+
+godebug default=go1.22
```

Comment:
move this up to be by the go directive, and I bet we'll have to bump the go directive to `go 1.23.0` for this to work

Reply:
trying it out

Reply:
the go-compatibility CI failures are expected for now, will need to run `hack/update-vendor.sh` to update vendor.txt

Reply:
will also need to update build/dependencies.yaml to the new go version since this PR is doing all the updates at once

Reply:
Is it okay to update dependencies.yaml without making rest of the runner changes? I'm not too familiar with this part sorry 
xref https://github.com/kubernetes/kubernetes/pull/127118

I've update it in the latest commit, but I'm guessing to change things like the kube-cross image, we will need to build it for the new go1.23.4 minor?